### PR TITLE
Fix Zero Bug #8

### PR DESCRIPTION
**Pull request by dns2utf8**
Implementing #8

### CODE REVIEW

**Comment by dns2utf8** [comment]
Does this fix all the cases you can think of?
@rnestler @dbrgn

File: src/main.rs [hunk 1]
```diff
@@ -90,7 +122,18 @@ mod test {
...
+
+    #[test]
+    fn test_parse_pacman_zero_output_0() {
+        assert_eq!(Some("1.10.0_patch1-1".into()), parse_pacman_output("hdf5 1.10.0_patch1-1"));
+    }
+
+    #[test]
+    fn test_parse_pacman_zero_output_1() {
+        assert_eq!(Some("4.7.0-2".into()), parse_pacman_output("usbip 4.7-2
+"));
```
Comment (rnestler) [inline src/main.rs]: Whats the point of this strange newline?

**Comment by rnestler** [comment]
I'm actually thinking about creating Rust bindings to [lipalpm](https://www.archlinux.org/pacman/libalpm.3.html) to query the database directly instead of parsing the output of pacman. I think we could avoid a lot of issues that way.

**Comment by dns2utf8** [comment]
Good point, I started a crate: https://github.com/dns2utf8/alpm-rs

File: Cargo.lock [hunk 2]
```diff
@@ -134,3 +286,40 @@ name = "winapi-build"
 version = "0.1.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
+[metadata]
```
Comment (dbrgn) [inline Cargo.lock]: Wow, is that a new feature? Cool.

File: src/main.rs [hunk 3]
```diff
@@ -1,18 +1,46 @@
...
-        .skip(1)
-        .next()
+macro_rules! t {
+    ( $e : expr ) => (
+        match $e {
+            Some(r) => r,
+            None => return None,
+        }
+    )
+}
```
Comment (dbrgn) [inline src/main.rs]: That's `try!` for an Option, right?

Comment (rnestler) [inline src/main.rs]: Kind of, yes.

**Comment by rnestler** [comment]
@dns2utf8 Are you still interested in that one? Could we maybe split the parsing fixes and the migration to libalpm?

**Comment by rnestler** [comment]
I'll close this for now, since it is stale since a long time and some of the ideas got merged independently. Feel free to reopen or create a new PR if you want to add the parser changes @dns2utf8 :slightly_smiling_face:

**Pull request closed without merging**